File: library/src/main/java/org/xmtp/android/library/libxmtp/InboxState.kt
package org.xmtp.android.library.libxmtp

import org.xmtp.android.library.InboxId
import uniffi.xmtpv3.FfiInboxState
import uniffi.xmtpv3.FfiSignatureKind

typealias SignatureKind = FfiSignatureKind

class InboxState(
        private val ffiInboxState: FfiInboxState,
) {
    val inboxId: InboxId
        get() = ffiInboxState.inboxId
    val identities: List<PublicIdentity>
        get() = ffiInboxState.accountIdentities.map { PublicIdentity(it) }

    val installations: List<Installation>
        get() = ffiInboxState.installations.map { Installation(it) }

    val recoveryPublicIdentity: PublicIdentity
        get() = PublicIdentity(ffiInboxState.recoveryIdentity)

    val creationSignatureKind: SignatureKind?
        get() = ffiInboxState.creationSignatureKind
}

File: sdks/ios/Sources/XMTPiOS/Libxmtp/InboxState.swift
//
//  InboxState.swift
//
//
//  Created by Naomi Plasterer on 8/21/24.
//

import Foundation

public typealias SignatureKind = FfiSignatureKind

public struct InboxState {
	var ffiInboxState: FfiInboxState

	public var inboxId: InboxId {
		ffiInboxState.inboxId
	}

	public var identities: [PublicIdentity] {
		ffiInboxState.accountIdentities.map { PublicIdentity(ffiPrivate: $0) }
	}

	public var installations: [Installation] {
		ffiInboxState.installations.map { Installation(ffiInstallation: $0) }
	}

	public var recoveryIdentity: PublicIdentity {
		PublicIdentity(ffiPrivate: ffiInboxState.recoveryIdentity)
	}

	/// The type of signature that was used to create the inbox initially.
	/// Future signatures from this identity must be of the same kind
	public var creationSignatureKind: SignatureKind? {
		ffiInboxState.creationSignatureKind
	}
}
